tests/parsers/test_google_decimeter.py

```diff
@@ -0,0 +1,823 @@
+    # csv file
+    np.testing.assert_equal(derived[row_name, index], value)
+
+
+def test_get_and_set_num(derived):
+    """Testing __getitem__ and __setitem__ methods for numerical values
+
+    Parameters
+    ----------
+    derived : pytest.fixture
+        Instance of AndroidDerived2021 for testing
+    """
+    key = 'testing123'
+    value = np.zeros(len(derived))
+    derived[key] = value
+    np.testing.assert_equal(derived[key, :], value)
+
+
+def test_get_and_set_str(derived):
+    """Testing __getitem__ and __setitem__ methods for string values
+
+    Parameters
+    ----------
+    derived : pytest.fixture
+        Instance of AndroidDerived2021 for testing
+    """
+    key = 'testing123_string'
+    value = ['word']*len(derived)
+    derived_size = len(derived)
+    size1 = int(derived_size/2)
+    size2 = (derived_size-int(derived_size/2))
+    value1 = ['ashwin']*size1
+    value2 = ['derek']*size2
+    value = np.concatenate((np.asarray(value1, dtype=object),
+                            np.asarray(value2, dtype=object)))
+    derived[key] = value
+
+    np.testing.assert_equal(derived[key, :], value)
+
+def test_android_concat(derived, pd_df):
+    """Test concat on Android data.
+
+    Parameters
+    ----------
+    derived : AndroidDerived2021
+        Instance of AndroidDerived2021 for testing
+    pd_df : pytest.fixture
+        pd.DataFrame for testing measurements
+    """
+
+    # remove first timestamp to match
+    pd_df = pd_df[pd_df['millisSinceGpsEpoch'] != pd_df.loc[0,'millisSinceGpsEpoch']]
+
+    # extract and combine gps and glonass data
+    gps_data = derived.where("gnss_id","gps")
+    glonass_data = derived.where("gnss_id","glonass")
+    gps_glonass_navdata = gps_data.concat(glonass_data)
+    glonass_gps_navdata = glonass_data.concat(gps_data)
+
+    # combine using pandas
+    gps_df = pd_df[pd_df["constellationType"]==1]
+    glonass_df = pd_df[pd_df["constellationType"]==3]
+    gps_glonass_df = pd.concat((gps_df,glonass_df))
+    glonass_gps_df = pd.concat((glonass_df,gps_df))
+
+    for combined_navdata, combined_df in [(gps_glonass_navdata, gps_glonass_df),
+                                          (glonass_gps_navdata, glonass_gps_df)]:
+
+        # check a few rows to make sure they're equal
+        np.testing.assert_array_equal(combined_navdata["raw_pr_m"],
+                                      combined_df["rawPrM"])
+        np.testing.assert_array_equal(combined_navdata["raw_pr_sigma_m"],
+                                      combined_df["rawPrUncM"])
+        np.testing.assert_array_equal(combined_navdata["intersignal_bias_m"],
+                                      combined_df["isrbM"])
+
+def test_navdata_type(derived):
+    """Test that all subclasses inherit from NavData
+
+    Parameters
+    ----------
+    derived : pytest.fixture
+        Instance of AndroidDerived2021 for testing
+    """
+    isinstance(derived, NavData)
```

Comment:
Is an assert statement needed here? If not, perhaps an assert statement would improve clarity?

Reply:
an assert statement _is_ needed there or else nothing happens. This test was effectively pointless. Updated 13 places throughout the repo